### PR TITLE
MINOR: fix typo of ProducerConfig and KafkaProducer

### PR DESCRIPTION
**Pull request by leo0842**
ADPATIVE -> ADAPTIVE

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java [hunk 1]
```diff
@@ -94,8 +94,8 @@ public class ProducerConfig extends AbstractConfig {
                                                  + "batch size is under this <code>batch.size</code> setting.";
 
     /** <code>partitioner.adaptive.partitioning.enable</code> */
-    public static final String PARTITIONER_ADPATIVE_PARTITIONING_ENABLE_CONFIG = "partitioner.adaptive.partitioning.enable";
```
Comment (divijvaidya) [inline clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java:97]: This is a public field in a public class: https://kafka.apache.org/35/javadoc/org/apache/kafka/clients/producer/ProducerConfig.html#PARTITIONER_ADPATIVE_PARTITIONING_ENABLE_CONFIG 

Many customers may have code which may be relying on this typo and this change will break those customers. There is very little upside in changing it now and huge downside wrt backward compatibility. I will suggest to keep it as it is for now. We will have to live with it until next major version (4.x).

Comment (C0urante) [reply clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java:97]: If we want to remove this field in a future version, we can deprecate it now and introduce a new field with the correct spelling.

That might fall into KIP territory, though. @divijvaidya thoughts?

Comment (divijvaidya) [reply clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java:97]: Thanks for your suggestion, that is the right way to go about it @C0urante. I understand that a KIP for this change might be a an overdo but let's follow the rules decided by the community (and create a KIP for deprecation).

**Comment by divijvaidya** [comment]
@leo0842 since this is your first contribution (thank you for your participation!), you may not be familiar with the concept of KIP. You can read more about it here: https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Improvement+Proposals

**Pull request closed without merging**